lib/doorkeeper/openid_connect.rb

```diff
@@ -26,11 +26,30 @@
 
 module Doorkeeper
   module OpenidConnect
-    # TODO: make this configurable
-    SIGNING_ALGORITHM = 'RS256'.freeze
+    def self.signing_algorithm
+      configuration.signing_algorithm.to_s.upcase.to_sym
+    end
 
     def self.signing_key
-      JSON::JWK.new(OpenSSL::PKey.read(configuration.jws_private_key))
+      key =
+        if [:HS256, :HS384, :HS512].include?(signing_algorithm)
+          configuration.signing_key
+        else
+          OpenSSL::PKey.read(configuration.signing_key)
+        end
+      JSON::JWK.new(key)
+    end
+
+    def self.signing_key_normalized
+      key = signing_key
+      case key[:kty].to_sym
+      when :RSA
+        key.slice(:kty, :kid, :e, :n)
+      when :EC
+        key.slice(:kty, :kid, :x, :y)
+      when :oct
+        key.slice(:kty, :kid, :k)
```

Comment:
Yeah, I think `oct` is the only symmetric one,  I'm a bit out of my wheelhouse as well. Check out section 6 [here](https://tools.ietf.org/html/rfc7518). Per [nov's comment](https://github.com/nov/json-jwt/issues/48#issuecomment-313528111), maybe it's smarter to just leave it out of the jwks response for now, and you could easily add it later based on need.